llvm/lib/Passes/PassBuilderPipelines.cpp

```diff
@@ -1985,9 +1986,23 @@ ModulePassManager PassBuilder::buildO0DefaultPipeline(OptimizationLevel Level,
 
   MPM.addPass(createModuleToFunctionPassAdaptor(AnnotationRemarksPass()));
 
+  if (PTO.OptimizeSYCLFramework) {
+    assert(Level == OptimizationLevel::O0 &&
+           "Optimization of sycl framework should be used only with O0 "
+           "optimization level");
```

Comment:
We are within `buildO0DefaultPipeline` and there is already assertion that optimization level is set to `-O0`: do we really need to duplicate it?